source/sigv4.c

```diff
@@ -246,6 +246,67 @@ static SigV4Status_t verifySigV4Parameters( const SigV4Parameters_t * pParams );
 static void hexEncode( SigV4String_t * pInputStr,
                        SigV4String_t * pHexOutput );
 
+/**
+ * @brief Extract each header key and value from passed pHeaders string.
+ *
+ * @param[in] pHeaders HTTP headers to canonicalize.
+ * @param[in] headersLen Length of HTTP headers to canonicalize.
+ * @param[in] flags Flag to indicate if headers are already
+ * in the canonical form.
+ * @param[out] canonicalRequest Struct to maintain intermediary buffer
+ * and state of canonicalization.
+ */
+static SigV4Status_t generateCanonicalHeaders( const char * pHeaders,
+                                               size_t headersLen,
+                                               uint32_t flags,
+                                               CanonicalContext_t * canonicalRequest );
+
+/**
+ * @brief Append Signed Headers to the string which needs to be signed.
```

Comment:
"Signed Headers" are not part of the "String to sign" according to the documentation: https://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html

Does this statement need to be updated?

Reply:
Signed Headers are part of canonical request which is turn are hashed and added to string we need to sign to get the auth signature.

Reply:
It's more appropriate to say that it is appended to the canonical request

Reply:
```suggestion
 * @brief Append "Signed Headers" to the Canonical Request buffer.
```